operator/webserver.py

```diff
@@ -41,33 +43,62 @@ def launch_simulator(self):
         if self.simulation_process is not None:
             abort(400, description='Simulation already running.')
 
+        print("Creating logfile for run")
+
         # Get team config
         teamId = request.json['id']
         self.mission = request.json['mission']
+        track = request.json['track']
         for obj in self.team_config['teams']: 
             if obj['id'] == teamId: self.team = obj  
 
+        # Create log file. Create logs directory if it does not exist
+        filename = 'logs/{}_{}_{}.txt'.format(self.team['name'].lower().replace(' ', '-'), self.mission, str(datetime.now().strftime("%d-%m-%Y_%Hh%Mm%Ss")))
+        if not os.path.exists(os.path.dirname(filename)):
+            try:
+                os.makedirs(os.path.dirname(filename))
+            except OSError as exc: # Guard against race condition
+                if exc.errno != errno.EEXIST:
+                    raise
+
+        # Write to log file
+        self.log_file = open(filename, 'w')
+        self.log_file.write('Launching simulator '+track+'...\n')
+
+        print("Launching simulator...")
+       
+
         # Write team specific car settings to settings.json
         filename = os.path.realpath(os.path.dirname(__file__)) + '/../settings.json'
         with open(filename, 'w') as file:
             json.dump(self.team['car_settings'], file, sort_keys=True, indent=4, separators=(',', ': '))
 
-        # Launch Unreal Engine simulator
-        self.simulation_process = subprocess.Popen(['../simulator/FSDS.exe'], creationflags=subprocess.CREATE_NEW_PROCESS_GROUP)
-        time.sleep(7)
+        try:
```

Comment:
surround this whole thing in try-catch so that if the launching fails the started process can be killed and the error can be shown to the user.